### PR TITLE
Gracefully fail when constant evaluation function can't be compiled or run

### PR DESCRIPTION
**Pull request by nickgg**
Summary: Coming back to the issue of constant folding: still having trouble with backend specific Nodes which are not implemented on the interpreter. This PR changes the behaviour so that if there is a terminal failure during CF it stops trying to constant fold. 

This usually isn't an issue in Glow since backend specific nodes are definitionally not standard and so don't often occur in input networks, but it comes up in unit tests for some backends.

Documentation: NFC

Test Plan: Ran tests in debug and release, ran my backends tests.

### CODE REVIEW

File: lib/Optimizer/GraphOptimizer/ConstantFolding.cpp [hunk 1]
```diff
@@ -311,7 +312,10 @@ bool glow::ConstantFold::run(Function *F, const CompilationContext &cctx) {
     }
 
     // Compute the constant value of the node.
-    std::vector<Constant *> constResults = constantFoldNodeImpl(*backend, N);
+    std::vector<Constant *> constResults;
```
Comment (opti-mix) [inline lib/Optimizer/GraphOptimizer/ConstantFolding.cpp]: Maybe use a SmallVector here to avoid heap allocations?

Comment (nickgg) [reply lib/Optimizer/GraphOptimizer/ConstantFolding.cpp]: We'd have to replace the vector everywhere in this file, but we could do it. I'm not sure theres a sensible number of results to use for the SmallVector though, it scales proportionally with the size of the network.

Comment (opti-mix) [reply lib/Optimizer/GraphOptimizer/ConstantFolding.cpp]: Yeah, let's keep it simple. Ignore my previous comment!

**Comment by nickgg** [comment]
Don't think that works - we already have a check on `isOpSupported()`. To trigger this case you need to find a case where you're compiling for backend A which supports all nodes in the graph, but the Interpreter does not. Can you think of a case where that could happen? No backend supports that Select node configuration for example.

**Comment by nickgg** [comment]
OK, having talked to @opti-mix and @jfix71 offline I have convinced them this is a) already covered implicitly and b) super complicated to test explicitly. I'll land and one day when im looking for good karma come back and write the mock backend that can handle this test.

**Pull request closed without merging**